python/graphstorm/eval/utils.py

```diff
@@ -304,7 +304,7 @@ def broadcast_data(rank, world_size, data_tensor):
     if get_backend() == "gloo":
         device = "cpu"
     elif get_backend() == "nccl":
-        assert data_tensor.is_cuda
```

Comment:
@zheng-da Do you have concerns about this change? I found data tensor is not on gpu in nccl

Reply:
the question is which GPU is used here?

Reply:
I think .cuda goes to the correpsonding rank of the device.